include/openssl/crypto.h.in

```diff
@@ -85,6 +85,14 @@ __owur int CRYPTO_THREAD_write_lock(CRYPTO_RWLOCK *lock);
 int CRYPTO_THREAD_unlock(CRYPTO_RWLOCK *lock);
 void CRYPTO_THREAD_lock_free(CRYPTO_RWLOCK *lock);
 
+# if defined(__GNUC__) && !defined(OPENSSL_NO_STDINT_H)
+typedef __attribute__((__aligned__(8))) uint64_t CRYPTO_ATOMIC_U64;
+#  define CRYPTO_ATOMIC_IS_ALIGNED_U64(val) (((uintptr_t)(val) & (8 - 1)) == 0)
+# else
```

Comment:
There should be made a way to opt this out, because this can make atomic operations slower for operands not aligned to 8-byte boundaries.